shared_model/cryptography/ed25519_sha3_impl/internal/ed25519_impl.cpp

```diff
@@ -29,17 +29,19 @@ namespace iroha {
              const pubkey_t &pub,
              const privkey_t &priv) {
     sig_t sig;
-    ed25519_sign(sig.data(), msg, msgsize, pub.data(), priv.data());
+    ed25519_sign(reinterpret_cast<signature_t *>(sig.data()),
+                 msg,
+                 msgsize,
+                 reinterpret_cast<const public_key_t *>(pub.data()),
+                 reinterpret_cast<const private_key_t *>(priv.data()));
     return sig;
   }
 
   sig_t sign(const std::string &msg,
              const pubkey_t &pub,
              const privkey_t &priv) {
-    sig_t sig;
-    ed25519_sign(
-        sig.data(), (uint8_t *)msg.data(), msg.size(), pub.data(), priv.data());
-    return sig;
+    return sign(
```

Comment:
Why make an indirect call when library call can be made directly as it was before?

Reply:
1. it increases coverage, because you need to cover only one function instead of N
2. for compiler and resulting binary there will be no any implications, because of optimizations